src/v/serde/serde.h

```diff
@@ -99,12 +99,17 @@ using serde_size_t = uint16_t;
 using serde_size_t = uint32_t;
 #endif
 
+template<class T, template<class, size_t> class C>
+struct is_specialization_of_sized : std::false_type {};
+template<template<class, size_t> class C, class T, size_t N>
```

Comment:
will this template match for any normal integer or only ones that are specifically `size_t` ?

Reply:
So maybe it would be better to use `std::enable_if<>` and add a match for `std::is_integral<>` or something of the sort